### PR TITLE
Fix issue running scons using a symlink to scons.py in an scons-local dir

### PR DESCRIPTION
**Pull request by mwichmann**
Earlier fix to running with scons-local when the directory is a
symlink (i.e. scons-local symlinked to scons-local-{versiontag} and
invoked as scons-local/scons.py) broke running scons when the
"executable" is a symlink. Restoring that behavior.

This is dodgy to test reliably on a developer system, as the
fallback mechansim to find a usable scons goes on to look for
a system-installed scons, so it looked like it was working. I
will add a test if I can figure out a good way.

Signed-off-by: Mats Wichmann <mats@linux.com>

## Contributor Checklist:

* [ ] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [X] I have updated `master/src/CHANGES.txt` directory (and read the `README.txt` in that directory)
* [X] I have updated the appropriate documentation

### CODE REVIEW

File: src/CHANGES.txt [hunk 1]
```diff
@@ -14,11 +14,14 @@ RELEASE  VERSION/DATE TO BE FILLED IN LATER
     - Improve finding of Microsoft compiler: add a 'products' wildcard
       in case 2017 Build Tools only is installed as it is considered a separate
       product from the default Visual Studio
+    - scons.py and sconsign.py stopped working if script called as a symlink
```
Comment (bdbaddog) [inline src/CHANGES.txt:55]: Change to "Fixed issue running scons-local when called via a symlink"

File: src/CHANGES.txt [hunk 2]
```diff
@@ -14,11 +14,14 @@ RELEASE  VERSION/DATE TO BE FILLED IN LATER
     - Improve finding of Microsoft compiler: add a 'products' wildcard
       in case 2017 Build Tools only is installed as it is considered a separate
       product from the default Visual Studio
+    - Fixed issue running scons-local using a symlink to the scons.py in
```
Comment (bdbaddog) [inline src/CHANGES.txt]: oh is there and issue #  for this, if so please reference?

Comment (mwichmann) [reply src/CHANGES.txt]: No, the report came in on irc and I don't see a GH issue for it.

**Comment by bdbaddog** [comment]
Can you make a test for this so we don't break it again going forward?

**Pull request merged**